tests/collections/collection.test.ts

```diff
@@ -383,6 +429,95 @@ for (const testClient in testClients) {
+        assert.strictEqual(_.keys(res.insertedIds).length, 20);
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        const updateManyResp = await collection.updateMany({"city": "la"},
+                              {
+                                "$set" : { "state" : "ca" }
+                              });     
+        assert.strictEqual(updateManyResp.matchedCount, 0);
+        assert.strictEqual(updateManyResp.modifiedCount, 0);
+        assert.strictEqual(updateManyResp.acknowledged, true);
+        assert.strictEqual(updateManyResp.upsertedCount, undefined);
+        assert.strictEqual(updateManyResp.upsertedId, undefined);
+      });
+      //TODO skipped until https://github.com/stargate/jsonapi/issues/273 is fixed
+      it.skip('should upsert with upsert flag set to true when not found', async () => {
+        let docList = Array.from({ length: 2 }, ()=>({username: "id", city : "nyc"}));
+        docList.forEach((doc, index) => {
+          doc.username = doc.username+(index+1);
+        });
+        const res = await collection.insertMany(docList);    
+        assert.strictEqual(res.insertedCount, docList.length);
+        assert.strictEqual(res.acknowledged, true);
+        assert.strictEqual(_.keys(res.insertedIds).length, 2);
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        const updateManyResp = await collection.updateMany({"city": "la"},
+                              {
+                                "$set" : { "state" : "ca" }
+                              },
+                              {
+                                "upsert": true
+                              });     
+        assert.strictEqual(updateManyResp.matchedCount, 0);
+        assert.strictEqual(updateManyResp.modifiedCount, 0);
+        assert.strictEqual(updateManyResp.acknowledged, true);
+        assert.strictEqual(updateManyResp.upsertedCount, 1);
+        assert.ok(updateManyResp.upsertedId);
+      });
+      it('should fail when moreData returned by updateMany as true', async () => {
+        let docList = Array.from({ length: 25 }, ()=>({username: "id", city : "nyc"}));
+        docList.forEach((doc, index) => {
+          doc.username = doc.username+(index+1);
+        });
+        const res = await collection.insertMany(docList);    
+        assert.strictEqual(res.insertedCount, docList.length);
+        assert.strictEqual(res.acknowledged, true);
+        assert.strictEqual(_.keys(res.insertedIds).length, 25);
+        
+        //const idToUpdateAndCheck = sampleDocsWithIdList[0]._id;
+        try{
+          const updateManyResp = await collection.updateMany({"city": "nyc"},
+                                                    {
+                                                      "$set" : { "state" : "ny" }
+                                                    });
+        }
+        catch(e: any){
+          assert.ok(e);
```

Comment:
Unfortunately, this test will pass if no error is thrown. There are a couple of patterns we recommend for asserting failures with async/await with JavaScript:

1) Golang-style error handling.

```
const err = await collection.updateMany(filter, update).then(() => null, err => err);
assert.ok(err);
assert.strictEqual(err.message, "...");
```

2) `assert.rejects()`

```javascript
await assert.rejects(collection.updateMany(filter, update), /More than 20 records found/);
```

3) Throw at the end of the `try` block

```javascript
try {
  await collection.updateMany(filter, update);
  assert.ok(false);
} catch (err) {
  // ...
}
```

(3) requires the least amount of changes from the current state of the code, but I find (1) to be the easiest one to work with. What do you think @kathirsvn ?

Reply:
Good catch @vkarpov15 I cleaned up, all the exception based tests in [this PR](https://github.com/stargate/stargate-mongoose/pull/55/files) I missed this and ll fix this as well. Basically, I used the 3rd option you mentioned above on that PR.